testdata/p4tc_samples_outputs/simple_extern_example_control_blocks.c

```diff
@@ -0,0 +1,342 @@
+    unsigned char ebpf_byte;
+    u32 pkt_len = skb->len;
+
+    struct my_ingress_metadata_t *meta;
+    hdrMd = BPF_MAP_LOOKUP_ELEM(hdr_md_cpumap, &ebpf_zero);
+    if (!hdrMd)
+        return TC_ACT_SHOT;
+    unsigned ebpf_packetOffsetInBits = hdrMd->ebpf_packetOffsetInBits;
+    hdr_start = pkt + BYTES(ebpf_packetOffsetInBits);
+    hdr = &(hdrMd->cpumap_hdr);
+    meta = &(hdrMd->cpumap_usermeta);
+{
+        struct p4tc_ext_bpf_params* ext_params;
+        struct p4tc_ext_bpf_val* ext_val;
+        u8 hit;
+        u32 val_0 = 0;
+        struct reg_val_t arg_val_0;
+        __builtin_memset((void *) &arg_val_0, 0, sizeof(struct reg_val_t ));
+        {
+            arg_val_0.protocol = hdr->ipv4.protocol;
+                        arg_val_0.aux = 22;
+            /* reg3_0.write(2, arg_val_0) */
+            __builtin_memset(ext_params, 0, sizeof(struct p4tc_ext_bpf_params));
+            ext_params->pipe_id = p4tc_filter_fields.pipeid;
+            ext_params->ext_id = 1;
+            ext_params->inst_id = 2;
+            ext_params->index = 2;
+
+            __builtin_memcpy(ext_val->out_params, &arg_val_0, sizeof(struct reg_val_t ));
+            bpf_p4tc_extern_md_write(skb, ext_params, sizeof(ext_params), ext_val, sizeof(ext_val));
+;
+            /* nh_table_0.apply() */
+            {
+                /* construct key */
+                struct p4tc_table_entry_act_bpf_params__local params = {
+                    .pipeid = p4tc_filter_fields.pipeid,
+                    .tblid = 1
+                };
+                struct ingress_nh_table_key key;
+                __builtin_memset(&key, 0, sizeof(key));
+                key.keysz = 32;
+                key.field0 = hdr->ipv4.srcAddr;
+                struct p4tc_table_entry_act_bpf *act_bpf;
+                /* value */
+                struct ingress_nh_table_value *value = NULL;
+                /* perform lookup */
+                act_bpf = bpf_p4tc_tbl_read(skb, &params, sizeof(params), &key, sizeof(key));
+                value = (struct ingress_nh_table_value *)act_bpf;
+                if (value == NULL) {
+                    /* miss; find default action */
+                    hit = 0;
+                } else {
+                    hit = value->hit;
+                }
+                if (value != NULL) {
+                    /* run action */
+                    switch (value->action) {
+                        case INGRESS_NH_TABLE_ACT_INGRESS_EXT_REG: 
+                            {
+                                __builtin_memset(ext_params, 0, sizeof(struct p4tc_ext_bpf_params));
+                                ext_params->pipe_id = p4tc_filter_fields.pipeid;
+                                ext_params->ext_id = 1;
+                                ext_params->inst_id = 1;
+                                ext_params->index = value->u.ingress_ext_reg.port_id;
+
+                                ext_val = bpf_p4tc_extern_md_read(skb, ext_params, sizeof(ext_params));
```

Comment:
For md_read you should also adapt to the pointer

```suggestion
                                ext_val = bpf_p4tc_extern_md_read(skb, ext_params, sizeof(*ext_params));
```